File: java/src/test/java/QuestionNineTest.java
import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.Test;

public class QuestionNineTest {
	private Node root;

	private int sumOfNodes = 0;

	@AfterEach
	void cleanUp() {
		root = null;
		sumOfNodes = 0;
	}

	@Test
	void testTreeSumIsEqualToExpectedSum() {
		prepareTree();
		assertThat(QuestionNine.sum(root)).isEqualTo(sumOfNodes);
	}

	private void prepareTree() {
		Node n11 = new Node(3);
		sumOfNodes += n11.val;
		Node n12 = new Node(3);
		sumOfNodes += n12.val;
		Node n13 = new Node(3);
		sumOfNodes += n13.val;
		Node n21 = new Node(3);
		sumOfNodes += n21.val;
		Node n22 = new Node(3);
		sumOfNodes += n22.val;
		Node n23 = new Node(3);
		sumOfNodes += n23.val;
		Node n1 = new Node(3, new Node[]{n11, n12, n13});
		Node n2 = new Node(3, new Node[]{n21, n22, n23});
		sumOfNodes += n1.val;
		sumOfNodes += n2.val;
		root = new Node(2, new Node[]{n1, n2});
		sumOfNodes += root.val;
	}
}

File: go/tests/question_nine_test.go
package test

import (
	"challenge/questions"
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestSumTree(t *testing.T) {
	sum, root := prepareTestData()
	assert.Equal(t, sum, questions.Sum(root))
}

func prepareTestData() (int, questions.Node) {
	sumOfNodes := 0
	n11 := questions.Node{Val: 3}
	sumOfNodes += n11.Val
	n12 := questions.Node{Val: 3}
	sumOfNodes += n12.Val
	n13 := questions.Node{Val: 3}
	sumOfNodes += n13.Val
	n21 := questions.Node{Val: 3}
	sumOfNodes += n21.Val
	n22 := questions.Node{Val: 3}
	sumOfNodes += n22.Val
	n23 := questions.Node{Val: 3}
	sumOfNodes += n23.Val
	n1 := questions.Node{Val: 3, Children: []questions.Node{n11, n12, n13}}
	n2 := questions.Node{Val: 3, Children: []questions.Node{n21, n22, n23}}
	sumOfNodes += n1.Val
	sumOfNodes += n2.Val
	root := questions.Node{Val: 2, Children: []questions.Node{n1, n2}}
	sumOfNodes += root.Val
	return sumOfNodes, root
}
